build-npm/action.yml

```diff
@@ -0,0 +1,87 @@
+name: Build NPM
+descripion: Build a npm-module
+
+inputs:
+  github_artifact_retention:
+    description: 'Time during which artifact are kept in Github infrastructure'
+    default: '2'
+  tests_module_path:
+    description: 'Path to a folder in the repository containing a tests module to be built'
+    required: false
+    default: 'tests/jahia-module/'   
+  node_version:
+    description: 'Version of node to install on the host'
+    required: false
+    default: 'lts/*'
+
+runs:
+  using: 'composite'
+  steps:
+    - uses: actions/setup-node@v3
+      with:
+        node-version: ${{ inputs.node_version }}
+
+    - name: Install jq
+      shell: bash
+      run: sudo apt-get install jq
+
+    - name: Get package.json infos
+      shell: bash
+      run: |
+        echo "MODULE_NAME=$(jq '.name' package.json | cut -d\" -f2 )" >> $GITHUB_ENV
+
+        MODULE_VERSION=$(jq '.version' package.json | cut -d\" -f2 )
+
+        if [[ $MODULE_VERSION == *-SNAPSHOT ]]; then
+          echo "MODULE_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+        else
+          echo "MODULE_OLD_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+          MODULE_VERSION+="-SNAPSHOT"
+          echo $MODULE_VERSION
+          echo "MODULE_VERSION=$MODULE_VERSION" >> $GITHUB_ENV
+        fi
+
+    - name: Prepare the module
+      shell: bash
+      run: yarn
```

Comment:
You should add a module_path parameter to allow the building of a module located in another directory than the repo's root